addons/sms/models/sms_sms.py

```diff
@@ -159,59 +158,49 @@ def _split_batch(self):
             yield sms_batch
 
     def _send(self, unlink_failed=False, unlink_sent=True, raise_exception=False):
-        """ This method tries to send SMS after checking the number (presence and
-        formatting). """
-        iap_data = [{
-            'res_id': record.id,
-            'number': record.number,
-            'content': record.body,
-        } for record in self]
+        """Send SMS after checking the number (presence and formatting)."""
+        messages = [{
+            'content': body,
+            'numbers': [{'number': record.number, 'uuid': record.uuid} for record in state_records],
+        } for body, state_records in tools.groupby(self, key=lambda record: record.body)]
```

Comment:
"state_records", "record", can't we find better names?

Also I see to remember a method to directly do browserecord.groupby('field'), without using tools.